### DISCUSSION TITLE
Server side system initiated database actions

### DISCUSSION THREAD

**Discussion by 24sheep** [pocketbase/pocketbase#1593] [Q&A] [answered]
If all my database interactions are server-side, should I be using app.Dao() for my database actions rather than pb.collection('~')? My app does lots of system actions where there is no @auth user trigger.  This means having to suspend all the API rules. I'm not using the API at all. So I'm thinking I have been going about my DB interactions all wrong. If I do need to use DBX are there any good examples of using it in a Sveltekit that anyone knows of? Thanks.

**Comment by ganigeorgiev** [comment]
I'm not entirely sure that I understand what you are asking.

At the moment you can interact with the database abstraction only if you [use PocketBase as framework](https://pocketbase.io/docs/use-as-framework/) and extend it with Go, aka. you can't use `app.Dao()` inside a JS application (_support for binding to the app event hooks with JS is not available yet but it is planned for v0.13/v0.14_).

If you want to interact with the PocketBase database from a 3rd party server-side application you have to use the [available web APIs](https://pocketbase.io/docs/api-records/) and eventually authorize as admin (collection API rules don't apply for admins and admins have access to all resources).
_Alternatively, you can access and modify directly the SQLite db file (`pb_data/data.db`), but keep in mind that the realtime events, schema constraint validations and other PocketBase specific things will not work out of the box._

**Comment by 24sheep** [comment, marked as answer]
I'm only interacting with the Database server side. So far I have been doing so through pb.collection() but I'm running into problems with the API rules when the system initiates a transaction since the system is not an @auth user. Is there a way to do non @auth actions in a secure way with the API rules in place?

**Reply by ganigeorgiev** [reply]
> Is there a way to do non @auth actions in a secure way with the API rules in place?

I'm sorry, I still don't understand what do you mean.

As mentioned in the previous comment, if you want to perform a server-side action ignoring the API rules, you can authorize as admin since they can read and modify everything, eg. something like:

```js
await pb.admins.authWithPassword('test@example.com', '1234567890');

console.log(pb.authStore.model)

await pb.collection('example1').create({...})
await pb.collection('example2').create({...})
```

**Reply by 24sheep** [reply]
I'm apologize for my lack of clarity. Despite that you got to the heart of my question. That should do the trick. Thank you.
